File: Graphs/Island and Treasure/Island.java
class Solution {
    public void islandsAndTreasure(int[][] grid) {
        int[][] directions = {{0,1},{0,-1},{1,0},{-1,0}};
        Deque<int[]> q = new ArrayDeque<>();
        int rows = grid.length;
        int cols = grid[0].length;


        for (int row = 0; row < rows; row++){
            for (int col = 0; col < cols; col++){
                if (grid[row][col] == 0){
                    q.add(new int[]{row, col});
                }
            }
        }

        while (!q.isEmpty()){
            int[] coords = q.poll();  

            for (int[] dir : directions) {
                int dr = dir[0];
                int dc = dir[1];
                
                int nr = dr + coords[0];
                int nc = dc + coords[1];

                if (nr >= 0 && nr < rows && nc >= 0 && nc < cols && grid[nr][nc] == 2147483647){
                    grid[nr][nc] = 1 + grid[coords[0]][coords[1]];
                    q.add(new int[] {nr,nc});
                }
            }
        }
    }
}

File: Graphs/Island and Treasure/Island.py
class Solution:
    def islandsAndTreasure(self, grid: List[List[int]]) -> None:
        rows,cols = len(grid), len(grid[0])
        directions = [[0,1],[0,-1],[1,0],[-1,0]]
        q = deque()

        for row in range(rows):
            for col in range(cols):
                if grid[row][col] == 0:
                    q.append((row,col))

        while q:
            row, col = q.popleft()

            for dr, dc in directions:
                nr, nc = dr + row, dc + col

                if 0 <= nr < rows and 0 <= nc < cols and grid[nr][nc] == 2147483647:
                    grid[nr][nc] = grid[row][col] + 1
                    q.append((nr, nc))

